test/server/cards/03_TWI/units/KrakenConfederateTactician.spec.ts

```diff
@@ -0,0 +1,66 @@
+describe('Kraken Confederate Tactician', function () {
+    integration(function (contextRef) {
+        it('Kraken Confederate Tactician\'s ability create 2 battle droid tokens when played and should give +1/+1 to friendly tokens unit on attack', function () {
+            contextRef.setupTest({
+                phase: 'action',
+                player1: {
+                    hand: ['drop-in', 'kraken#confederate-tactician'],
+                    groundArena: ['clone-trooper'],
+                },
+                player2: {
+                    groundArena: ['battlefield-marine', 'battle-droid'],
+                }
+            });
+
+            const { context } = contextRef;
+
+            // We play Kraken, it creates 2 battle droids
+            context.player1.clickCard(context.kraken);
+
+            let battleDroids = context.player1.findCardsByName('battle-droid');
+            expect(battleDroids.length).toBe(2);
+            expect(battleDroids).toAllBeInZone('groundArena');
+            expect(battleDroids.every((battleDroid) => battleDroid.exhausted)).toBeTrue();
+            expect(battleDroids.every((battleDroid) => battleDroid.getHp() === 1)).toBeTrue();
+            expect(battleDroids.every((battleDroid) => battleDroid.getPower() === 1)).toBeTrue();
+            context.moveToNextActionPhase();
+
+            // Kraken attack gives +1/+1 to all friendly tokens
+            context.player1.clickCard(context.kraken);
+            context.player1.clickCard(context.p2Base);
+            const cloneTrooper = context.player1.findCardsByName('clone-trooper')[0];
+            expect(cloneTrooper.getPower()).toBe(3);
+            expect(cloneTrooper.getHp()).toBe(3);
+
+            // Opponent's token units are not impacted
+            const opponentBattleDroids = context.player2.findCardsByName('battle-droid');
+            expect(opponentBattleDroids.length === 1).toBeTrue();
+            expect(opponentBattleDroids[0].getPower()).toBe(1);
+            expect(opponentBattleDroids[0].getHp()).toBe(1);
+
+            battleDroids = context.player1.findCardsByName('battle-droid');
+            expect(battleDroids.every((battleDroid) => battleDroid.getHp() === 2)).toBeTrue();
+            expect(battleDroids.every((battleDroid) => battleDroid.getPower() === 2)).toBeTrue();
+
+            context.player2.clickCard('battlefield-marine');
+            context.player2.clickCard(cloneTrooper);
+
+            // But newly created token units are not impacted
+            context.player1.clickCard(context.dropIn);
+            const cloneTroopers = context.player1.findCardsByName('battle-droid');
```

Comment:
Bug here - you're searching by `'battle-droid'`

Reply:
Thanks for catching it !